sklearn/ensemble/_bagging.py

```diff
@@ -380,6 +370,7 @@ def fit(self, X, y, *, sample_weight=None, **fit_params):
         self : object
             Fitted estimator.
         """
+        sample_weight = fit_params.pop("sample_weight", None)
         _raise_for_params(fit_params, self, "fit")
```

Comment:
a better alternative here would be to allow for `_raise_for_params(fit_params, self, "fit", allow=["sample_weight"])` so that we don't have to `pop` from the dict, and then put it back later, which is the patter in all methods here.

Reply:
Done